### ISSUE TITLE
It doesn't seem to properly handle JSX (tsx) if jsx is set to "preserve".

### ISSUE THREAD

**Issue by XanderEmu** [TypeStrong/tsify#175]
I'm trying to use tsify with grunt, browserify and babel.

My Gruntfile.js looks like this:
```js
module.exports = function(grunt) {

    // Project configuration.
    grunt.initConfig({
        pkg: grunt.file.readJSON('package.json'),
        browserify: {
            options: {
                browserifyOptions: {
                    debug: true
                }
            },
            dist: {
                files: {
                    'build/app.js': [ 'src/app.tsx' ]
                },
                options: {
                    plugin: [
                        ['tsify', {
                            target: 'es6',
                            removeComments: true,
                            preserveConstEnums: true,
                            jsx: 'preserve'
                        }]
                    ],
                    transform: [
                        ['babelify', {
                            sourceMap: true,
                            presets: ['es2015'],
                            plugins: [
                                ['incremental-dom', {
                                    prefix: 'IncrementalDOM'
                                }]
                            ]
                        }]
                    ]
                }
            }
        }
    });

    //
    grunt.loadNpmTasks('grunt-browserify');

    //
    grunt.registerTask('build', ['browserify']);

    // Default task(s).
    grunt.registerTask('default', ['build']);

}
```

I have some JSX (not React!!!) code in my source. I get a compilation error from it:

 return <div class="foo">
>>                ^
>> ParseError: Unexpected token

TSX code itself is fine. I compiled it using separate grunt-ts and it worked properly.

**Comment by cartant** [comment]
Is the `es2015` Babel preset sufficient to process JSX? I wouldn't have thought so. If the JSX is making its way past Babel to Browserify, that could be the problem - Browserify won't like it.

**Comment by XanderEmu** [comment]
There is. See my dependencies:

```json
{
  "dependencies": {
    "babel-plugin-incremental-dom": "^3.4.0",
    "babel-preset-es2015": "^6.9.0",
    "babelify": "^7.3.0",
    "grunt": "^1.0.1",
    "grunt-browserify": "^5.0.0",
    "incremental-dom": "^0.4.1",
    "redux": "^3.5.2",
    "tsify": "^1.0.3",
    "typescript": "^1.8.10",
    "typings": "^1.3.2"
  }
}

```

**Comment by cartant** [comment]
Ah, I see; it's the `incremental-dom` Babel plugin that does the transformation.

I think the only thing you are missing is the `extensions` option for `babelify`:

    ['babelify', {
        sourceMap: true,
        presets: ['es2015'],
        plugins: [
            ['incremental-dom', {
                prefix: 'IncrementalDOM'
            }]
        ],
        extensions: ['.tsx']
    }]

**Comment by XanderEmu** [comment]
Nope, that's not it.

**Comment by cartant** [comment]
That is really strange - and unfortunate. To look into your problem, I messed with `tsify`'s [JSX example](https://github.com/TypeStrong/tsify/tree/master/examples/jsx) and changed its Browserify-API-based bundling to the following:

    browserify()
        .add('src/main.tsx')
        .plugin(tsify, {
            jsx: 'preserve',
            target: 'ES6'
        })
        .transform(babelify, {
            extensions: ['.tsx'],
            presets: ['es2015'],
            plugins: [
                ['incremental-dom', {
                    prefix: 'IncrementalDOM'
                }]
            ]
        })
        .bundle()
        .on('error', function (error) { console.error(error.toString()); })
        .pipe(process.stdout);

And that works for me. There's no syntax error and it outputs a sensible bundle. Perhaps it might be worthwhile running something like the above - as a test - to take `grunt-browserify` out of the loop?

**Comment by cartant** [comment]
I've found your problem. It's `grunt-browserify`. It [loads the transforms first and then the plugins](https://github.com/jmreidy/grunt-browserify/blob/master/lib/runner.js#L99-L109), so what you want to do - put the transform after the plugin - is simply not possible.

The simplest solution will probably be to add a `grunt-browserify` [`configure` function](https://github.com/jmreidy/grunt-browserify#configure) and to set up the plugin and transform (as done in my API-based example above) inside said function. I've not done that for a while, but remember it as being pretty straightforward.

Closing this as it's not something that can be addressed in `tsify`.
